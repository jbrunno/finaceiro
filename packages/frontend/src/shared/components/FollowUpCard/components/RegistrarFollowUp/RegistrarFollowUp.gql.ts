import { gql } from '@apollo/client';

export const REGISTRAR_FOLLOWPUP_MUTATION = gql`
  mutation RegistrarClienteFollowUp($input: RegistrarClienteFollowUpInput!) {
    registrarClienteFollowUp(input: $input) {
      id
    }
  }
`;

export type RegistrarClienteFollowUpInput = {
  clienteId: string;
  tituloId: string;
  situacaoFollowUpId: string;
  descricao: string;
};

export type RegistrarClienteFollowUpData = {
  id: string;
};
