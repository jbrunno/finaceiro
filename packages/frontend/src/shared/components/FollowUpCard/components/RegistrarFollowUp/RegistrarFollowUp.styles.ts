import { css, styled } from '@frontend/styles';

export const StyledForm = styled('form')(
  ({ theme }) => css`
    display: grid;
    row-gap: ${theme.fn.spacing(16)};
    width: 100%;
    min-width: 25rem;
  `,
);
