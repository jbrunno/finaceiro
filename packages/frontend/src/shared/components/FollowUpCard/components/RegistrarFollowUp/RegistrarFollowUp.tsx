import React, { Dispatch, SetStateAction } from 'react';
import { Button, TextField, Typography } from '@frontend/components';
import { AutocompleteSituacaoFollowUp } from '@/shared/components/AutocompleteSituacaoFollowUp/AutocompleteSituacaoFollowUp';
import { FormFields, useRegistarFollowUp } from './hooks/useRegistrarFollowUp';
import { StyledForm } from './RegistrarFollowUp.styles';
import { SituacoesFollowUp } from '../../FollowUpCard.gql';

export type RegistrarFollowUpProps = {
  tituloId?: string;
  situacoes: SituacoesFollowUp[] | undefined;
  setSituacoesKeyword: Dispatch<SetStateAction<string | undefined>>;
};

export function RegistrarFollowUp({
  tituloId,
  setSituacoesKeyword,
  situacoes,
}: RegistrarFollowUpProps) {
  const { formik, situacao } = useRegistarFollowUp(tituloId);

  return (
    <StyledForm onSubmit={formik.handleSubmit}>
      <Typography variant="subtitle1">Registrar Follow</Typography>
      <AutocompleteSituacaoFollowUp
        label="Situação do Follow-up"
        initialValue={situacao}
        filterSituacoes={setSituacoesKeyword}
        situacoes={situacoes && situacoes.sort((a, b) => a.codigo - b.codigo)}
        onChange={(value) => {
          formik.handleChange(FormFields.situacao)(value?.id || '');
        }}
      />
      <TextField
        id="descricaoFollow"
        label="Descreva o Follow-up"
        variant="outlined"
        size="small"
        value={formik.values[FormFields.descricao]}
        onChange={formik.handleChange(FormFields.descricao)}
        errorText={formik.errors[FormFields.descricao]}
        isTouched={formik.touched[FormFields.descricao]}
      />

      <Button
        type="submit"
        variant="contained"
        color="primary"
        size="large"
        disabled={!formik.values[FormFields.situacao]}
      >
        Registrar Follow
      </Button>
    </StyledForm>
  );
}
