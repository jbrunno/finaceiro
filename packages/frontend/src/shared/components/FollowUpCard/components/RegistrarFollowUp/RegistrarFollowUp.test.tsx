import { render } from '@test-utils/component-testing';
import { RegistrarFollowUp } from './RegistrarFollowUp';

test('should render RegistrarFollowUp', () => {
  const { container } = render(
    <RegistrarFollowUp situacoes={[]} setSituacoesKeyword={jest.fn} />,
  );
  expect(container).toMatchSnapshot();
});
