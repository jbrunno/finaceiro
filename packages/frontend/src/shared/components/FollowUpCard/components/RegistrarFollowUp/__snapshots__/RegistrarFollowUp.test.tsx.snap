// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RegistrarFollowUp 1`] = `
.emotion-0 {
  display: grid;
  row-gap: 1rem;
  width: 100%;
  min-width: 25rem;
}

.emotion-2 {
  margin: 0;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
}

.emotion-3.Mui-focused .MuiAutocomplete-clearIndicator {
  visibility: visible;
}

@media (pointer: fine) {
  .emotion-3:hover .MuiAutocomplete-clearIndicator {
    visibility: visible;
  }
}

.emotion-3 .MuiAutocomplete-tag {
  margin: 3px;
  max-width: calc(100% - 6px);
}

.emotion-3 .MuiAutocomplete-inputRoot {
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.MuiAutocomplete-hasPopupIcon.emotion-3 .MuiAutocomplete-inputRoot,
.MuiAutocomplete-hasClearIcon.emotion-3 .MuiAutocomplete-inputRoot {
  padding-right: 30px;
}

.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon.emotion-3 .MuiAutocomplete-inputRoot {
  padding-right: 56px;
}

.emotion-3 .MuiAutocomplete-inputRoot .MuiAutocomplete-input {
  width: 0;
  min-width: 30px;
}

.emotion-3 .MuiInput-root {
  padding-bottom: 1px;
}

.emotion-3 .MuiInput-root .MuiInput-input {
  padding: 4px 4px 4px 0px;
}

.emotion-3 .MuiInput-root.MuiInputBase-sizeSmall .MuiInput-input {
  padding: 2px 4px 3px 0;
}

.emotion-3 .MuiOutlinedInput-root {
  padding: 9px;
}

.MuiAutocomplete-hasPopupIcon.emotion-3 .MuiOutlinedInput-root,
.MuiAutocomplete-hasClearIcon.emotion-3 .MuiOutlinedInput-root {
  padding-right: 39px;
}

.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon.emotion-3 .MuiOutlinedInput-root {
  padding-right: 65px;
}

.emotion-3 .MuiOutlinedInput-root .MuiAutocomplete-input {
  padding: 7.5px 4px 7.5px 5px;
}

.emotion-3 .MuiOutlinedInput-root .MuiAutocomplete-endAdornment {
  right: 9px;
}

.emotion-3 .MuiOutlinedInput-root.MuiInputBase-sizeSmall {
  padding-top: 6px;
  padding-bottom: 6px;
  padding-left: 6px;
}

.emotion-3 .MuiOutlinedInput-root.MuiInputBase-sizeSmall .MuiAutocomplete-input {
  padding: 2.5px 4px 2.5px 8px;
}

.emotion-3 .MuiFilledInput-root {
  padding-top: 19px;
  padding-left: 8px;
}

.MuiAutocomplete-hasPopupIcon.emotion-3 .MuiFilledInput-root,
.MuiAutocomplete-hasClearIcon.emotion-3 .MuiFilledInput-root {
  padding-right: 39px;
}

.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon.emotion-3 .MuiFilledInput-root {
  padding-right: 65px;
}

.emotion-3 .MuiFilledInput-root .MuiFilledInput-input {
  padding: 7px 4px;
}

.emotion-3 .MuiFilledInput-root .MuiAutocomplete-endAdornment {
  right: 9px;
}

.emotion-3 .MuiFilledInput-root.MuiInputBase-sizeSmall {
  padding-bottom: 1px;
}

.emotion-3 .MuiFilledInput-root.MuiInputBase-sizeSmall .MuiFilledInput-input {
  padding: 2.5px 4px;
}

.emotion-3 .MuiInputBase-hiddenLabel {
  padding-top: 8px;
}

.emotion-3 .MuiFilledInput-root.MuiInputBase-hiddenLabel {
  padding-top: 0;
  padding-bottom: 0;
}

.emotion-3 .MuiFilledInput-root.MuiInputBase-hiddenLabel .MuiAutocomplete-input {
  padding-top: 16px;
  padding-bottom: 17px;
}

.emotion-3 .MuiFilledInput-root.MuiInputBase-hiddenLabel.MuiInputBase-sizeSmall .MuiAutocomplete-input {
  padding-top: 8px;
  padding-bottom: 9px;
}

.emotion-3 .MuiAutocomplete-input {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-overflow: ellipsis;
  opacity: 1;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-5 {
  color: #3B3B4199;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(100% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(14px, 9px) scale(1);
  -moz-transform: translate(14px, 9px) scale(1);
  -ms-transform: translate(14px, 9px) scale(1);
  transform: translate(14px, 9px) scale(1);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: none;
}

.emotion-5.Mui-focused {
  color: #4A31FF;
}

.emotion-5.Mui-disabled {
  color: #3B3B4161;
}

.emotion-5.Mui-error {
  color: #F44336;
}

.emotion-6 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  color: #3B3B41DE;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  border-radius: 4px;
  padding-right: 14px;
}

.emotion-6.Mui-disabled {
  color: #3B3B4161;
  cursor: default;
}

.emotion-6:hover .MuiOutlinedInput-notchedOutline {
  border-color: #3B3B41DE;
}

@media (hover: none) {
  .emotion-6:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-6.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #4A31FF;
  border-width: 2px;
}

.emotion-6.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #F44336;
}

.emotion-6.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: #00000042;
}

.emotion-7 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-top: 1px;
  padding: 8.5px 14px;
  padding-right: 0;
}

.emotion-7::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7:focus {
  outline: 0;
}

.emotion-7:invalid {
  box-shadow: none;
}

.emotion-7::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-7.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: #3B3B4161;
}

.emotion-7:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-7:-webkit-autofill {
  border-radius: inherit;
}

.emotion-8 {
  position: absolute;
  right: 0;
  top: calc(50% - 14px);
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: #0000008A;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 2px;
  margin-right: -2px;
}

.emotion-9::-moz-focus-inner {
  border-style: none;
}

.emotion-9.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-9 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-9:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-9:hover {
    background-color: transparent;
  }
}

.emotion-9.Mui-disabled {
  background-color: transparent;
  color: #00000042;
}

.emotion-10 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-11 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-12 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-13 {
  float: unset;
  width: auto;
  overflow: hidden;
  display: block;
  padding: 0;
  height: 11px;
  font-size: 0.75em;
  visibility: hidden;
  max-width: 0.01px;
  -webkit-transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  white-space: nowrap;
}

.emotion-13>span {
  padding-left: 5px;
  padding-right: 5px;
  display: inline-block;
  opacity: 0;
  visibility: visible;
}

.emotion-14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-16 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  color: #3B3B41DE;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 4px;
}

.emotion-16.Mui-disabled {
  color: #3B3B4161;
  cursor: default;
}

.emotion-16:hover .MuiOutlinedInput-notchedOutline {
  border-color: #3B3B41DE;
}

@media (hover: none) {
  .emotion-16:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-16.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #4A31FF;
  border-width: 2px;
}

.emotion-16.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #F44336;
}

.emotion-16.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: #00000042;
}

.emotion-17 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-top: 1px;
  padding: 8.5px 14px;
}

.emotion-17::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-17::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-17:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-17::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-17:focus {
  outline: 0;
}

.emotion-17:invalid {
  box-shadow: none;
}

.emotion-17::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-17.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: #3B3B4161;
}

.emotion-17:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-17:-webkit-autofill {
  border-radius: inherit;
}

.emotion-21 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.9375rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 8px 22px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #4A31FF;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  text-transform: none;
}

.emotion-21::-moz-focus-inner {
  border-style: none;
}

.emotion-21.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-21 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-21:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #2025EF;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-21:hover {
    background-color: #4A31FF;
  }
}

.emotion-21:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-21.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-21.Mui-disabled {
  color: #00000042;
  box-shadow: none;
  background-color: #0000001F;
}

<div>
  <form
    class="emotion-0 emotion-1"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 emotion-2"
    >
      Registrar Follow
    </h6>
    <div
      class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon emotion-3"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-4"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined emotion-5"
          data-shrink="false"
          for="autocomplete-outlined"
          id="autocomplete-outlined-label"
        >
          Situação do Follow-up
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot emotion-6"
        >
          <input
            aria-autocomplete="both"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused emotion-7"
            id="autocomplete-outlined"
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <div
            class="MuiAutocomplete-endAdornment emotion-8"
          >
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator emotion-9"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-10"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root emotion-11"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline emotion-12"
          >
            <legend
              class="emotion-13"
            >
              <span>
                Situação do Follow-up
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root emotion-14"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined emotion-5"
        data-shrink="false"
        for="descricaoFollow"
        id="descricaoFollow-label"
      >
        Descreva o Follow-up
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall emotion-16"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall emotion-17"
          id="descricaoFollow"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline emotion-12"
        >
          <legend
            class="emotion-13"
          >
            <span>
              Descreva o Follow-up
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge emotion-20 emotion-21"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      Registrar Follow
    </button>
  </form>
</div>
`;
